### PR TITLE
Js courses

### PR DESCRIPTION
**Pull request by JustynaSmoliga**
Hi, I have added features:
- course list on courses menu tab
- course edit 
- course delete
- course create

### CODE REVIEW

File: frontend/src/components/CourseList/CourseListSlice.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      name: courseDto.name,
+      description: courseDto.description,
+      endDate: new Date(Date.parse(courseDto.endDate)), 
+      startDate: new Date(Date.parse(courseDto.startDate)),
+    };
+    return courseListElement;
+  });
+  courses.sort(function (courseListElement1, courseListElement2) {
+    var courseListElementStartDate1 = courseListElement1.startDate.getTime();
+    var courseListElementStartDate2 = courseListElement2.startDate.getTime();
```
Comment (robert-czwartosz) [inline frontend/src/components/CourseList/CourseListSlice.ts:43]: It would be better to use `const` instead of `var`.

**Review by robert-czwartosz — COMMENTED** [review]
Component looks great!
Adding section to course doesn't work.... after click ADD it moves me to /sections.

File: frontend/src/api/courses.service.ts [hunk 2]
```diff
@@ -1,16 +1,49 @@
...
+    return this.httpService.post(this.endpoint, course);
+  }
+
+  async fetchCourses() {
+    return this.httpService.get(this.endpoint);
+  }
+
+  async fetchCourse(courseId:string){
+    return this.httpService.get(`${this.endpoint}/${courseId}`);
+  }
```
[the remark this answers is not in the document]
Comment (JustynaSmoliga) [reply frontend/src/api/courses.service.ts]: Done, I removed fetchCourse ;)

File: frontend/src/components/Course/CourseDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+              variant="outlined"
+              multiline
+              onChange={handleDescriptionChange}
+              value={description}
+            ></TextField>
+          ) : (
+            <Box>
+              <p>{description}</p>
+            </Box>
+          )}
```
[the remark this answers is not in the document]
Comment (JustynaSmoliga) [reply frontend/src/components/Course/CourseDetails.tsx]: Done

File: frontend/src/components/CourseList/CourseListSlice.ts (hunk 1 above)
Comment (JustynaSmoliga) [reply frontend/src/components/CourseList/CourseListSlice.ts]: Done

File: frontend/src/app/ActiveCourse.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+export function getActiveCourse(): CourseListElementModel | undefined{
+const activeCourseItem = localStorage.getItem("activeCourse");
+if(!activeCourseItem){
+    return undefined;
+}
+return JSON.parse(activeCourseItem);
+}
+
```
[the remark this answers is not in the document]
Comment (JustynaSmoliga) [reply frontend/src/app/ActiveCourse.ts]: Done, I have moved it to utils.ts file.

**Comment by JustynaSmoliga** [comment]
We have talked about ADD button on one of our meetings and we have agreed, that this button must redirect to create section screen. When sections will be merged, we will add suitable path.

**Review by robert-czwartosz — APPROVED** [review]

**Review by skorpiom — APPROVED** [review]
Looks and works great!

**Pull request merged**